### PR TITLE
chore: Send params to TAAR (fix #3804)

### LINKED ISSUE #3804: [TAAR] Pass locale/platform to the TAAR-api when opening the Discopane
This is a feature request for the [TAAR project](https://github.com/mozilla/taar/). When Firefox opens the Discopane, it sends along some bit of information (locale, platform, ...). 

We would love this informations to be propagated to the TAAR api to make use of it if some user requesting recommendations is not in our database.

I will file a similar issue in mozilla/addons-server.

### PR DESCRIPTION
**Pull request by tofumatt**
Fix #3804.

These tests are quite old so this patch is pretty meh... I didn't want to spend a lot of time fixing up the tests because it'd be a PITA to review and I need to start on the user pages... there aren't actually many more weeks to this quarter 😓

The lang is already sent to the API so this just adds the platform sent to the Disco Pane to the API request, which addons-server will send to TAAR.

### CODE REVIEW

File: tests/unit/disco/test_actions.js [hunk 1]
```diff
@@ -41,6 +45,13 @@ describe('disco/actions/getDiscoResults', () => {
       .toThrow(/errorHandlerId is required/);
   });
 
+  it('requires platform', () => {
+    const params = defaultParams();
+    delete params.platform;
+    expect(() => getDiscoResults(params))
```
Comment (willdurand) [inline tests/unit/disco/test_actions.js]: I thought we had to make things a little bit more explicit? Something like:

```
expect(() => {
  // func
}).toThrow();
```

File: tests/unit/disco/test_api.js [hunk 2]
```diff
@@ -17,23 +17,30 @@ describe(__filename, () => {
 
   describe('getDiscoveryAddons', () => {
     it('calls the API', () => {
-      getDiscoveryAddons({ api: apiState });
+      getDiscoveryAddons({ api: apiState, platform: 'Windows' });
```
Comment (willdurand) [inline tests/unit/disco/test_api.js]: Do we have constants for that? Windows, Darwin, etc.

File: src/disco/containers/DiscoPane.js [hunk 3]
```diff
@@ -176,8 +184,9 @@ export function loadedAddons(state) {
   );
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state, ownProps) {
   return {
+    platform: ownProps.params.platform,
```
Comment (willdurand) [inline src/disco/containers/DiscoPane.js]: I dont see the `params` prop in the `PropTypes`.

File: tests/unit/disco/containers/TestDiscoPane.js [hunk 4]
```diff
@@ -203,12 +213,15 @@ describe(__filename, () => {
     it('gets discovery results when results are empty', () => {
       const dispatch = sinon.stub();
       const errorHandler = new ErrorHandler({ id: 'some-id', dispatch });
-      const props = helpers.mapStateToProps(loadDiscoResultsIntoState([]));
+      const props = helpers.mapStateToProps(
+        loadDiscoResultsIntoState([]), ownProps);
 
-      render({ errorHandler, dispatch, ...props });
+      render({ errorHandler, dispatch, platform: 'Darwin', ...props });
```
Comment (willdurand) [inline tests/unit/disco/containers/TestDiscoPane.js]: Why is this needed if `props` is constructed using `ownProps` before?

Comment (tofumatt) [reply tests/unit/disco/containers/TestDiscoPane.js]: Oh, possibly a holdover. I think it can be removed.

File: tests/unit/disco/test_actions.js (hunk 1 above)
Comment (tofumatt) [reply tests/unit/disco/test_actions.js]: Indeed! This was copypasta. I'll fix this one and the one I copied it from.

File: src/disco/containers/DiscoPane.js (hunk 3 above)
Comment (tofumatt) [reply src/disco/containers/DiscoPane.js]: Ah, I put `platform` in there because it's what `mapStateToProps` returns. Should I add `params` as well? That's confusing ^_^

Comment (willdurand) [reply src/disco/containers/DiscoPane.js]: Yeah, `params` are provided by react-router (like `location`). We need to declare all props, not only the mapped ones (that are redux only).

File: tests/unit/disco/test_api.js (hunk 2 above)
Comment (tofumatt) [reply tests/unit/disco/test_api.js]: We don't, they're different from what we usually use–I think Firefox sends them.

File: src/disco/containers/DiscoPane.js (hunk 3 above)
Comment (tofumatt) [reply src/disco/containers/DiscoPane.js]: Ah okay, cool.

File: tests/unit/disco/test_api.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/disco/test_api.js]: oh ok, fine then

**Comment by tofumatt** [comment]
Okay, I think I made all the tweaks needed for this.

File: tests/unit/disco/test_api.js (hunk 2 above)
Comment (tofumatt) [reply tests/unit/disco/test_api.js]: I don't even know all the values though, it would be great to document them going forward. I'll at least add a note about that...

File: src/disco/containers/DiscoPane.js [hunk 5]
```diff
@@ -33,6 +33,8 @@ export class DiscoPaneBase extends React.Component {
     i18n: PropTypes.object.isRequired,
     location: PropTypes.object.isRequired,
     mozAddonManager: PropTypes.object,
+    params: PropTypes.object.isRequired,
```
Comment (willdurand) [inline src/disco/containers/DiscoPane.js]: Well, ideally you want to be more explicit:

```
params: {
  platform: PropTypes.string.isRequired,
},
```

File: src/disco/containers/DiscoPane.js [hunk 6]
```diff
@@ -33,6 +33,8 @@ export class DiscoPaneBase extends React.Component {
...
+    platform: PropTypes.string.isRequired,
```
Comment (willdurand) [inline src/disco/containers/DiscoPane.js]: Aaaah I see. I think you dont need to do that. Instead, rely on `params.platform` in the code.

File: src/disco/containers/DiscoPane.js [hunk 7]
```diff
@@ -176,8 +185,9 @@ export function loadedAddons(state) {
   );
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state, ownProps) {
   return {
+    platform: ownProps.params.platform,
```
Comment (willdurand) [inline src/disco/containers/DiscoPane.js]: Yeah sorry, actually you should not do that. Remove the `ownProps` arg here, and use `params.platform` in the component.

File: tests/unit/disco/containers/TestDiscoPane.js [hunk 8]
```diff
@@ -126,14 +129,21 @@ describe(__filename, () => {
   });
 
   describe('mapStateToProps', () => {
+    it('sets platform', () => {
+      const props = helpers.mapStateToProps(
+        loadDiscoResultsIntoState([]), ownProps);
+
+      expect(props.platform).toEqual('Darwin');
+    });
```
Comment (willdurand) [inline tests/unit/disco/containers/TestDiscoPane.js]: So yeah, this should not be done here I guess. We should not rely on `ownProps` here.

Comment (willdurand) [reply tests/unit/disco/containers/TestDiscoPane.js]: The `params` prop should be added to the `renderProps()` function, example: https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestCollection.js#L45-L48.

File: tests/unit/disco/containers/TestDiscoPane.js [hunk 9]
```diff
@@ -241,7 +247,7 @@ describe(__filename, () => {
           guid: '@guid1',
           slug: 'discovery-addon-1',
         },
-      }]));
+      }]), { params: { platform: 'Darwin' } });
```
Comment (willdurand) [inline tests/unit/disco/containers/TestDiscoPane.js]: mmh this looks weird here. What is tested?

Comment (tofumatt) [reply tests/unit/disco/containers/TestDiscoPane.js]: Oops it's wrong and should have been removed but was before I was just adding `ownProps` everywhere. I'll remove it 😄
